src/Processor/ShippingMethodProcessor.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FriendsOfSylius\SyliusImportExportPlugin\Processor;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ShippingMethodInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Shipping\Model\ShippingCategoryInterface;
+
+final class ShippingMethodProcessor implements ResourceProcessorInterface
+{
+    /** @var FactoryInterface */
+    private $shippingMethodFactory;
+
+    /** @var RepositoryInterface */
+    private $shippingMethodRepository;
+
+    /** @var RepositoryInterface */
+    private $zoneRepository;
+
+    /** @var RepositoryInterface */
+    private $categoryRepository;
+
+    /** @var RepositoryInterface */
+    private $channelRepository;
+
+    /** @var ObjectManager */
+    private $manager;
+
+    /** @var MetadataValidatorInterface */
+    private $metadataValidator;
+
+    /** @var string[] */
+    private $headerKeys;
+
+    public function __construct(
+        FactoryInterface $shippingMethodFactory,
+        RepositoryInterface $shippingMethodRepository,
+        RepositoryInterface $zoneRepository,
+        RepositoryInterface $categoryRepository,
+        RepositoryInterface $channelRepository,
+        ObjectManager $manager,
+        MetadataValidatorInterface $metadataValidator,
+        array $headerKeys
+    ) {
+        $this->shippingMethodFactory = $shippingMethodFactory;
+        $this->shippingMethodRepository = $shippingMethodRepository;
+        $this->zoneRepository = $zoneRepository;
+        $this->categoryRepository = $categoryRepository;
+        $this->channelRepository = $channelRepository;
+        $this->manager = $manager;
+        $this->metadataValidator = $metadataValidator;
+        $this->headerKeys = $headerKeys;
+    }
+
+    public function process(array $data): void
+    {
+        $this->metadataValidator->validateHeaders($this->headerKeys, $data);
+
+        /** @var ShippingMethodInterface $shippingMethod */
+        $shippingMethod = $this->getShippingMethod($data['Code']);
+        $shippingMethod->setName($data['Name']);
+        $shippingMethod->setZone($this->findZone($data['Zone']));
+        $shippingMethod->setCategory($this->findCategory($data['Category']));
+        $shippingMethod->setCalculator($data['Calculator']);
+        $shippingMethod->setEnabled($data['Enabled']);
+        $shippingMethod->setDescription($data['Description']);
+        $shippingMethod->setConfiguration($data['Configuration']);
+        $shippingMethod->setPosition($data['Position']);
+        $shippingMethod->setCategoryRequirement($data['CategoryRequirement']);
```

Comment:
The existing convention in ImportExport plugin is to name use underscores and lowercase apart from the first letter.

```suggestion
        $shippingMethod->setCategoryRequirement($data['Category_requirement']);
```

Needs to be changed in more than this place though.

Reply:
Hmm, it looks like that convention is broken lol.